init.lua

```diff
@@ -89,18 +142,25 @@ local function ButtonGUI()
     -- Set this way up here so the theme can override.
     ImGui.PushStyleColor(ImGuiCol.ButtonHovered, 0.9, 0.9, 0.9, 0.5)
     ImGui.PushStyleColor(ImGuiCol.HeaderHovered, 0.9, 0.9, 0.9, 0.0)
-    for hotbarId, bmHotbar in ipairs(BMHotbars) do
-        local flags = ImGuiWindowFlags.NoFocusOnAppearing
-        if BMSettings:GetCharacterWindow(hotbarId).HideTitleBar then
-            flags = bit32.bor(flags, ImGuiWindowFlags.NoTitleBar)
-        end
-        if BMSettings:GetCharacterWindow(hotbarId).Locked then
-            flags = bit32.bor(flags, ImGuiWindowFlags.NoMove, ImGuiWindowFlags.NoResize)
-        end
-        if BMSettings:GetCharacterWindow(hotbarId).HideScrollbar then
-            flags = bit32.bor(flags, ImGuiWindowFlags.NoScrollbar)
+    if BMHotbars ~= nil then
+        for hotbarId, bmHotbar in ipairs(BMHotbars) do
+            if BMSettings:GetCharacterWindow(hotbarId) ~= nil then

```

Comment:
we should reduce this to a single lookup

local charWindowSettings =  BMSettings:GetCharacterWindow(hotbarId)

etc...